File: src/main/java/br/com/kronos/backend/aplicacao/contrato/Parcela.java
package br.com.kronos.backend.aplicacao.contrato;

import static br.com.kronos.backend.aplicacao.util.DecimalUtil.converterDoubleDoisDecimais;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Parcela {
    
    private long id;
    private Integer numero;
    private LocalDate dataVencimento;
    private Double valorOriginal;
    private LocalDate dataPagamento;
    private Double valorJuros;
    private Double valorMulta;
    private Double valorDesconto;
    private Double valorPagamento;
    private String observacao;
    private Integer idTipoFormaPagamento;
    private Long idContrato;

	@Builder
    public Parcela(long id, Integer numero, LocalDate dataVencimento, Double valorOriginal, LocalDate dataPagamento, Double valorJuros, 
                   Double valorMulta, Double valorDesconto, Double valorPagamento, String observacao, 
                   Integer idTipoFormaPagamento, Long idContrato) {

		super();
        this.id = id;
        this.numero = exigirNaoNulo(numero, "Número");  
        this.dataVencimento = exigirNaoNulo(dataVencimento, "data do vencimento");  
        exigirNaoNulo(valorOriginal, "Valor original");
        this.valorOriginal = valorOriginal != null ? converterDoubleDoisDecimais(valorOriginal) : 0.0;
        this.dataPagamento = dataPagamento;
        this.valorJuros = valorJuros != null ? converterDoubleDoisDecimais(valorJuros) : 0.0;
        this.valorMulta = valorMulta != null ? converterDoubleDoisDecimais(valorMulta) : 0.0; 
        this.valorDesconto = valorDesconto != null  ? converterDoubleDoisDecimais(valorDesconto) : 0.0;
        //TODO colocar a regra para gerar o valor de pagamento
        this.valorPagamento = valorPagamento != null ? converterDoubleDoisDecimais(valorPagamento) : 0.0;
        this.observacao = observacao;
        this.idTipoFormaPagamento = exigirNaoNulo(idTipoFormaPagamento, "Forma de pagamento");
        this.idContrato = exigirNaoNulo(idContrato, "Contrato"); 
	}
}

File: poc-kronos-frontend/src/model/parcela/Parcela.js
class Parcela {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.numero = dados ? dados.numero : null;
    this.dataVencimento = dados ? dados.dataVencimento : null;
    this.dataPagamento = dados ? dados.dataPagamento : null;
    this.valorJuros = dados ? dados.valorJuros : null;
    this.valorMulta = dados ? dados.valorMulta : null;
    this.valorDesconto = dados ? dados.valorDesconto : null;
    this.valorPagamento = dados ? dados.valorPagamento : null;
    this.valorOriginal = dados ? dados.valorOriginal : null;
    this.observacao = dados ? dados.observacao : null;
    this.idTipoFormaPagamento = dados ? dados.idTipoFormaPagamento : null;
    this.nomeTipoFormaPagamento = dados ? dados.nomeTipoFormaPagamento : null;
    this.idContrato = dados ? dados.idContrato : null;
    this.numeroParcelas = dados ? dados.numeroParcelas : null;
  }

  static validations(required, numeric, decimal) {
    return {
      numero: { required, numeric },
      dataVencimento: { required },
      valorOriginal: { required, decimal },
      valorJuros: { required, decimal },
      valorMulta: { required, decimal },
      valorDesconto: { required, decimal },
      valorPagamento: { required, decimal },
      idTipoFormaPagamento: { required },
      idContrato: { required }
    };
  }

  static validationMessages() {
    return {
      numero: {
        required: "Número é obrigatório",
        numeric: "Somente números"
      },
      dataVencimento: {
        required: "Data Vencimento é obrigatória"
      },
      valorOriginal: {
        required: "Valor Original é obrigatório",
        decimal: "Somente valor decimal"
      },
      valorJuros: {
        decimal: "Somente valor decimal"
      },
      valorMulta: {
        decimal: "Somente valor decimal"
      },
      valorDesconto: {
        decimal: "Somente valor decimal"
      },
      valorPagamento: {
        decimal: "Somente valor decimal"
      },
      idTipoFormaPagamento: {
        required: "Forma de pagamento é obrigatória"
      },
      idContrato: {
        required: "Contrato é obrigatório"
      }
    };
  }
}
export default Parcela;
